### ISSUE TITLE
Where is the complete documentation, including dos and donts for Angular.json ?

### ISSUE THREAD

**Issue by fmorriso** [angular/angular-cli#11556]
### Bug Report or Feature Request (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x ] feature request
```

### Area
```
- [ x] devkit
- [ ] schematics
```

### Versions
node: 8.9.2
npm: 6.2.0

### Repro steps
ng serve --extract-css

### The log given by the failure
Unknown option -extractCSS

### Desired functionality
A comprehensive guide (not just a syntax diagram) of how to set up angular.json, especially now that easy-to-use things such as --extract-css and others are no longer allowed during ng serve.  I don't mind that they are gone -- it is the lack of comprehensive "how to" (what is allowed with what, what is NOT allowed with what) inside angular.json.

I have searched in vain through the docs folder on GitHub.com, but cannot find what I need.

For example, how do I set up a "development" serve with the equivalent of the old "ng serve --common-chunk --vendor-chunk --extract-css  --named-chunks --watch"?   I cannot find this information anywhere.  Every blog article and ng-conf session I've looked at glosses over the changes, but doesn't provide links to a comprehensive guide on how to use angular.json to replace what we used to do via package.json.

**Comment by alan-agius4** [comment]
Documentation for angular workspace https://github.com/angular/angular-cli/wiki/angular-workspace

**Comment by fmorriso** [comment]
That WiKi page barely skims the surface in terms of answering my question.

Perhaps I was not clear and/or precise in what I was asking, so let me try again.

What are the Angulat-CLI 6.x Angular.json equivalents of the following Angular-CLI 1.7.x `ng build` and `ng serve` command-line arguments?
```
--extract-css
--common-chunk
--named-chunks
--vendor-chunk
```
In other words, I’m looking for a concise, **side-by-side “mapping”** of the old way to the new way of specifying the above options that used to be allowed directly on the ng command line.

Also, while I am at it, if the RxJS folks could find a way to create RxJS-compat to help ease the pain of moving from RxJS 5.x to RxJS 6.x,, why on Earth couldn’t the Angular-CLI crew come up a similar approach to ease the pain of moving from Angular-CLI 1.7.x to Angular-CLI 6.x?

**Comment by alan-agius4** [comment]
The CLI command is now avaiable at https://angular.io/cli and https://angular.io/guide/workspace-config
